### DISCUSSION TITLE
concourse upgrade from v6.7.4 to v7.4.3

### DISCUSSION THREAD

**Discussion by saketlj** [concourse/concourse#7866] [Help & Support]
Hi,

While upgrading concourse from v6.7.4 to v7.4.3, I am facing issue with web node, web process on web vm is not coming and failing with below error.

+ chown vcap:vcap /var/vcap/jobs/web/config/env/CONCOURSE_TSA_HOST_KEY
+ chmod 0600 /var/vcap/jobs/web/config/env/CONCOURSE_TSA_HOST_KEY
panic: descriptor Desc{fqName: "concourse_error_logs", help: "Number of error logged", constLabels: {}, variableLabels: [message]} is invalid: "bosh-deployment" is not a valid label name for metric "concourse_error_logs"

is env CONCOURSE_TSA_HOST_KEY introduced recently and Do i need to upgrade to any intermittent version before going with 7.4.3.

Thanks.

**Comment by taylorsilva** [comment]
`CONCOURSE_TSA_HOST_KEY` is a pretty old key. How did you initiate the upgrade?

**Reply by saketlj** [reply]
Yes, validated with old version and CONCOURSE_TSA_HOST_KEY is old.

We pull out software release from https://bosh.io/releases/github.com/concourse/concourse-bosh-release?version=7.1.0 ( e.g. for 7.1.0) and then deploy with bosh utility ( bosh -e ENV -d DEPLOYMENT deploy manifest ) designed in  bash script.

Although yesterday I was able to upgrade to v7.0.0 from v6.7.4 and planning to do 7.1, 7.2, 7.3 stepwise.

**Reply by saketlj** [reply]
I was able to successfully upgrade to v7.3.2 from v6.7.4. 

But when I am trying to move from 7.* to 7.4 still I am having above issue and it seems some issue with v7.4 versions.

**Reply by saketlj** [reply]
@taylorsilva 
Still I have difficulties to upgrade to 7.4 version and web process is not coming up, I did some troubleshooting with error message. I was able to bring the web node up by commenting out the following line

/var/vcap/jobs/web/config# vi bpm.yml
CONCOURSE_METRICS_ATTRIBUTE: "bosh-deployment:concourse,bosh-job:web"

P.S As upgrade ended in failure whenever I try to start/restart/, it fall back to previous version.

**Reply by saketlj** [reply]
@taylorsilva Do you have any update on this ?

**Reply by taylorsilva** [reply]
What errors are you getting now?

**Reply by saketlj** [reply]
@taylorsilva When I try to upgrade to v7.4 , web process doesn't come up with below error. 

```
panic: descriptor Desc{fqName: "concourse_error_logs", help: "Number of error logged", constLabels: {}, variableLabels: [message]} is invalid: "bosh-deployment" is not a valid label name for metric "concourse_error_logs"
[Uploading web-stderr-log.txt…]()
```

Now when I comment out below line from bpm.yml , I am able to bring up web process.

```
/var/vcap/jobs/web/config# vi bpm.yml
#CONCOURSE_METRICS_ATTRIBUTE: "bosh-deployment:concourse,bosh-job:web"
```

**Reply by taylorsilva** [reply]
Looks like that error is coming directly from prometheus. And looks like we've been using those labels for a long time: https://github.com/concourse/concourse-bosh-release/blob/40267e4676e9191610cbb451defaa2853dc9c6f4/jobs/web/templates/bpm.yml.erb#L22

I'm having trouble finding out if promethus put some restrictions on how labels are to be written 🤔 
https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details
Seems like convention is to use underscores instead of dashes as word separators in your labels.

@clarafu @xtremerui This may be a bug in the bosh release. See my first link. I'm having trouble finding prometheus docs that say "don't use dashes in your label names", so I'm not 100% sure about this 😕

**Reply by saketlj** [reply]
@clarafu @xtremerui  - do you have any update for me ?

**Reply by ywei2017** [reply]
@taylorsilva  - you are absolutely correct. I did a manual hack to change from "-" to "_", the web node was able to start successfully. I can also see the data in prometheus.

We can put in a PR to correct it if that helps.

Thanks
Yansheng

**Reply by xtremerui** [reply]
@ywei2017 a PR is welcome for sure.

**Reply by ywei2017** [reply]
@xtremerui just submitted [PR 164](https://github.com/concourse/concourse-bosh-release/pull/164)
